rpcs3/BasicKeyboardHandler.cpp

```diff
@@ -18,94 +17,109 @@ void BasicKeyboardHandler::Init(const u32 max_connect)
 	m_info.status[0] = CELL_KB_STATUS_CONNECTED; // (TODO: Support for more keyboards)
 }
 
-BasicKeyboardHandler::BasicKeyboardHandler()
+BasicKeyboardHandler::BasicKeyboardHandler(QObject* target, QObject* parent) : QObject(parent), m_target(target)
+{
+	// Adds event filter to the target to filter keyevents.
+	target->installEventFilter(this);
+}
+
+bool BasicKeyboardHandler::eventFilter(QObject* target, QEvent* ev)
 {
-	wxGetApp().Bind(wxEVT_KEY_DOWN, &BasicKeyboardHandler::KeyDown, this);
-	wxGetApp().Bind(wxEVT_KEY_UP, &BasicKeyboardHandler::KeyUp, this);
+	// Commenting target since I don't know how to target game window yet.
+	//if (target == m_target)
+	{
+		if (ev->type() == QEvent::KeyPress)
+		{
+			keyPressEvent(static_cast<QKeyEvent*>(ev));
+		}
+		else if (ev->type() == QEvent::KeyRelease)
+		{
+			keyReleaseEvent(static_cast<QKeyEvent*>(ev));
+		}
+	}
+	return false;
 }
 
-void BasicKeyboardHandler::KeyDown(wxKeyEvent& event)
+void BasicKeyboardHandler::keyPressEvent(QKeyEvent* keyEvent)
 {
-	Key(event.GetKeyCode(), 1);
-	event.Skip();
+	Key(keyEvent->key(), 1);
 }
 
-void BasicKeyboardHandler::KeyUp(wxKeyEvent& event)
+void BasicKeyboardHandler::keyReleaseEvent(QKeyEvent* keyEvent)
 {
-	Key(event.GetKeyCode(), 0);
-	event.Skip();
+	Key(keyEvent->key(), 0);
 }
 
 void BasicKeyboardHandler::LoadSettings()
 {
 	// Meta Keys
-	m_keyboards[0].m_buttons.emplace_back(WXK_CONTROL, CELL_KB_MKEY_L_CTRL);
-	m_keyboards[0].m_buttons.emplace_back(WXK_SHIFT, CELL_KB_MKEY_L_SHIFT);
-	m_keyboards[0].m_buttons.emplace_back(WXK_ALT, CELL_KB_MKEY_L_ALT);
-	m_keyboards[0].m_buttons.emplace_back(WXK_WINDOWS_LEFT, CELL_KB_MKEY_L_WIN);
-	m_keyboards[0].m_buttons.emplace_back(WXK_COMMAND, CELL_KB_MKEY_L_WIN);
+	m_keyboards[0].m_buttons.emplace_back(Qt::Key_Control, CELL_KB_MKEY_L_CTRL);
+	m_keyboards[0].m_buttons.emplace_back(Qt::Key_Shift, CELL_KB_MKEY_L_SHIFT);
+	m_keyboards[0].m_buttons.emplace_back(Qt::Key_Alt, CELL_KB_MKEY_L_ALT);
+	////m_keyboards[0].m_buttons.emplace_back(WXK_WINDOWS_LEFT, CELL_KB_MKEY_L_WIN); //TODO?
```

Comment:
Thanks, I'll commit that now..  If you see anything else (I'm sure there's plenty that'll be revised on this massive PR!), let me know.